core/src/main/scala/org/apache/spark/rdd/RDDCheckpointData.scala

```diff
@@ -130,5 +135,21 @@ private[spark] class RDDCheckpointData[T: ClassTag](@transient rdd: RDD[T])
   }
 }
 
-// Used for synchronization
-private[spark] object RDDCheckpointData
+private[spark] object RDDCheckpointData {
+  def rddCheckpointDataPath(sc: SparkContext, rddId: Int): Option[Path] = {
+    if (sc.checkpointDir.isDefined) {
+      Some(new Path(sc.checkpointDir.get, "rdd-" + rddId))
+    } else {
+      None
+    }
+  }
+
+  def clearRDDCheckpointData(sc: SparkContext, rddId: Int, blocking: Boolean = true) = {
+    rddCheckpointDataPath(sc, rddId).foreach { path =>
```

Comment:
We don't need the `rddCheckpointDataPath` method here. Instead we can just put it in a val.
```
val checkpointPath = sc.checkpointDir.map { d => new Path(p, "rdd-" + rddId) }
val fs = path.getFileSystem(...)
...
```